plugins/optimization-detective/optimization.php

```diff
@@ -172,6 +172,31 @@ function od_construct_preload_links( array $lcp_elements_by_minimum_viewport_wid
 	return implode( '', $preload_links );
 }
 
+/**
+ * Determines whether the response has an HTML Content-Type.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @return bool Whether Content-Type is HTML.
+ */
+function od_is_response_html_content_type(): bool {
+	$is_html_content_type = false;
+
+	$headers_list = array_merge(
+		array( 'Content-Type: ' . ini_get( 'default_mimetype' ) ),
+		headers_list()
+	);
+	foreach ( $headers_list as $header ) {
+		$header_parts = preg_split( '/\s*[:;]\s*/', strtolower( $header ) );
```

Comment:
Splitting on `:` and `;` here will result in `Content-Type: text/html; charset=utf-8` being split as `[ 'content-type', 'text/html', 'charset=utf-8' ]`. We then only use the first two items in the array.